lib/Doctrine/KeyValueStore/Storage/MongoDbStorage.php

```diff
@@ -30,57 +31,23 @@
 class MongoDbStorage implements Storage
 {
     /**
-     * @var \Mongo
+     * @var Client
      */
-    protected $mongo;
+    private $client;
 
     /**
-     * @var array
+     * @var string
      */
-    protected $dbOptions;
+    private $database;
 
     /**
-     * @var \MongoCollection
+     * @param Client $client
+     * @param string $database
      */
-    protected $collection;
-
-    /**
-     * Constructor
-     *
-     * @param \Mongo $mongo
-     * @param array  $dbOptions
-     */
-    public function __construct(\Mongo $mongo, array $dbOptions = [])
+    public function __construct(Client $client, $database)
```

Comment:
Since the database name is not really needed, how about changing the constructor signature to accept a `MongoDB\Database` object and letting people fetch it themselves?
```suggestion
    public function __construct(Database $client)
```

This makes storing the client and database name obsolete in favour of storing the database object.